### PR TITLE
Added Bedrock Cohere Embeddings Provider

### LINKED ISSUE #2475: Add Bedrock Cohere Embedding Model
### Validations

- [X] I believe this is a way to improve. I'll try to join the [Continue Discord](https://discord.gg/NWtdYexhMs) for questions
- [x] I'm not able to find an [open issue](https://github.com/continuedev/continue/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) that requests the same enhancement

### Problem

Currently Cohere embedding models are not supported through bedrock client. It would be great to have cohere embedding model along side current amazon.titan-embed-text models

### Solution

Add a new BedrockCohereEmbeddingsProvider to the existing embeddings. Only request and response needs different data structures.

### PR DESCRIPTION
**Pull request by Anantha-Kandrapu**
## Description

1. I added a Bedrock cohere embeddings provider. 
2. I had to add apache arrow as dependency because there were build issues without that package as vectorDB depends on it. (I pasted the error log in the issue).

## Checklist

- [x] The base branch of this PR is `dev`, rather than `main`
- [x] The relevant docs, if any, have been updated or created

## Screenshots

1. There's no visual changes.

## Testing
1) I have tested this change in debug mode as mentioned in the documentation, it was able to use bedrock cohere embedding model and fetch the results.
2) I ran the `npm run package` , copied the extension to another machine and tried it. It ran successfully.

## Fixes Apache-arrow error:
```
detail: undefined, id: '', location: { column: 31, file: '../../core/node_modules/vectordb/dist/remote/index.js', length: 14, line: 19, lineText: 'const apache_arrow_1 = require("apache-arrow");', namespace: '', suggestion: '' }, notes: [ { location: null, text: 'You can mark the path "apache-arrow" as external to exclude it from the bundle, which will remove this error. You can also surround this "require" call with a try/catch block to handle this failure at run-time instead of bundle-time.' } ], pluginName: '', text: 'Could not resolve "apache-arrow"' }, { detail: undefined, id: '', location: { column: 31, file: '../../core/node_modules/vectordb/dist/sanitize.js', length: 14, line: 24, lineText: 'const apache_arrow_1 = require("apache-arrow");', namespace: '', suggestion: '' }, notes: [ { location: null, text: 'You can mark the path "apache-arrow" as external to exclude it from the bundle, which will remove this error. You can also surround this "require" call with a try/catch block to handle this failure at run-time instead of bundle-time.' } ], pluginName: '', text: 'Could not resolve "apache-arrow"' } ], warnings: [] }

```

## Fix
This PR  fix #2475

### CODE REVIEW

**Comment by Anantha-Kandrapu** [comment]
I will squash and merge.

**Comment by sestinj** [comment]
@Anantha-Kandrapu thanks for the PR! It looks like a lot of the logic here might overlap with the original Bedrock provider—do you think that we could maintain just a single bedrock embeddings provider that can also support cohere?

**Comment by sestinj** [comment]
Also, I would prefer if the adding of the dependency were in a separate PR, as I'll probably want to review that a bit more closely and it isn't strictly necessary for this update

File: core/indexing/embeddings/BedrockCohereEmbeddingsProvider.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      modelId: this.options.model,
+      accept: "*/*",
+      contentType: "application/json"
+    };
+  }
+
+  private async _getCredentials() {
+    try {
+      return await fromIni({
+        profile: this.profile
```
Comment (chezsmithy) [inline core/indexing/embeddings/BedrockCohereEmbeddingsProvider.ts]: Check the most recent bedrock code and disable caching.

**Review by chezsmithy — CHANGES_REQUESTED** [review]

**Comment by Anantha-Kandrapu** [comment]
> @Anantha-Kandrapu thanks for the PR! It looks like a lot of the logic here might overlap with the original Bedrock provider—do you think that we could maintain just a single bedrock embeddings provider that can also support cohere?

I have also added a interface so that multiple other embedding providers from bedrock can be added in future.

**Review by chezsmithy — APPROVED** [review]

**Comment by Anantha-Kandrapu** [comment]
Hey @chezsmithy , Thanks for the approval earlier.
My fork was out of sync. There's one more approval I need. 
Thanks.

**Comment by Patrick-Erichsen** [comment]
Thanks for the great work here @Anantha-Kandrapu !

**Comment by sestinj** [comment]
Hi @Anantha-Kandrapu, yesterday we shared [some updates](https://discordapp.com/channels/1108621136150929458/1131313996750917835/1371996789967687783) with our contributors about how we're aiming to improve the contribution process. Part of this included the addition of a [Contributor License Agreement](https://github.com/continuedev/continue/blob/main/docs/docs/CLA.md) (CLA) to protect both contributors and the project. We're reaching out to ask that previous contributors sign it.

Could you please take a moment to sign, or if you have any questions send me a message? (either here or nate@continue.dev would work)

To do so, you just need to **post a comment below with the following text:**<br/>
```
I have read the CLA Document and I hereby sign the CLA
```

❤️ Thank you for the work you've done on Continue, and let me know if you have any suggestions on how we can make the project even better!

**Pull request merged**